packages/flutter/lib/src/material/text_field.dart

```diff
@@ -68,8 +68,22 @@ class TextField extends StatefulWidget {
   /// The [maxLines] property can be set to null to remove the restriction on
   /// the number of lines. By default, it is one, meaning this is a single-line
   /// text field. [maxLines] must not be zero. If [maxLines] is not one, then
-  /// [keyboardType] is ignored, and the [TextInputType.multiline] keyboard type
-  /// is used.
+  /// [keyboardType] is ignored, and the [TextInputType.multiline] keyboard
+  /// type is used.
+  ///
+  /// The [maxLength] property is set to null by default, which means the
+  /// number of characters allowed in the text field is not restricted. If
+  /// [maxLength] is set, a character counter will be displayed below the
+  /// field, showing how many characters have been entered and how many are
+  /// allowed. After [maxLength] characters have been input, additional input
+  /// is ignored, unless [maxLengthEnforced] is set to false. The TextField
+  /// enforces the length with a [LengthLimitingTextInputFormatter], which is
+  /// evaluated after the supplied [inputFormatters], if any. The [maxLength]
+  /// value must be either null or greater than zero.
+  ///
+  /// If [maxLengthEnforced] is set to false, then more than [maxLength]
+  /// characters may be entered, and the error counter and divider will
+  /// switch to the [decoration.errorStyle] when the limit is exceeded.
   ///
   /// The [keyboardType], [textAlign], [autofocus], [obscureText], and
   /// [autocorrect] arguments must not be null.
```

Comment:
add:

```
/// See also:
///
///  * [maxLength], which discusses the precise meaning of "number of characters" and how
///    it may differ from the intuitive meaning.
```